Update org owners, define authority and responsibilities (#37)
nix-open-org/org-v2: doc/github-org-owners.md

## GitHub org owners

> The following people are owners of the GitHub Organisation:

Reviewer: ```suggestion
The following people have the GitHub "owners" permissions on the NixOS organization:
```
Try to distinguish between real ownership and the github permissions. I expect this team to have some autonomy to unblock simple things, but overall follow the decisions of the steering comittee.

<!-- Keep this in sync with the code owners of github.md! -->
- [@infinisil](https://github.com/infinisil)
- [@lassulus](https://github.com/lassulus)
- [@tomberek](https://github.com/tomberek)
- [@winterqt](https://github.com/winterqt)
- [@zimbatm](https://github.com/zimbatm)

> 

Reviewer: ```suggestion
## How to contact the team
```

For any GitHub-related needs, you can reach out to the org owners by either:
- [Creating an issue in this repository](https://github.com/NixOS/org/issues/new).

> - Messaging in the [Github org owners help desk Matrix room](https://matrix.to/#/#org_owners:nixos.org).

Reviewer: Maybe something for later:

I'd like us to standardize using GitHub issues to discuss GitHub-related topics. Spreading information in two places is not optimal for transparency reasons.

### Authority and processes

> 

Reviewer: This section sounds procedural. It might make sense to add a more fuzzy higher-level description to make this less rigid.

Something like:
```suggestion
This team's role is to manage and unblock users of the github.com/NixOS GitHub organization. The @NixOS/steering gives them autonomy to handle small day-to-day tasks and expects them to escalate bigger decisions.
```

All org owners can individually take care of implementing:
- Decisions by bodies that have the authority to make GitHub org changes such as:
  - Arbitrary decisions by the Steering Committee.
  - Moderation decisions by the moderation team.
  - Changes from approved RFCs.
- Low-impact changes, such as:
  - Adding new org members to allow review requests.
  - Creating new unprivileged Nixpkgs teams for mention.
  - Updating repository meta information.
  - Taking care of occasional maintenance of otherwise unmaintained repositories.

With approval by at least 2 org owners, they can take care of implementing
higher-impact changes that are _not controversial_, such as:
- Administer unmaintained repos, such as:
  - Giving commit access to trusted people that offer maintenance.
  - Archiving if appropriate.

> - Create GitHub apps to unblock automation.

Reviewer 2: Could you expand on why you think this requires two+ approvals, but maintaining random unkempt repositories doesn't? (I'm not opposed, just trying to see if there's something I missed.)

I guess an app could be dangerous?

Author: The intent is that owners would only do maintenance with _low impact_ by themselves. It's nested under the "low-impact changes" point to imply that, but maybe that's not clear enough.

However, it's hard to estimate impact, so maybe err on the safe side and have this require 2+ approvals. Especially also because if necessary, 2+ org owners can delegate one of their own to have commit access to a specific repo and therefore empower them to maintain on their own.  

I'll move it for now, I convinced myself :)

Org owners have no authority to make other changes.

### Responsibilities

- Receive and process requests.

>   - Each request should land in at least 2 org owners inboxes of some kind.

Reviewer: ```suggestion
- Receive and process requests.
  - We expect org owners to be subscribed to the NixOS/org repo.
  - Each request should land in at least 2 org owners inboxes of some kind.
```

Author: In the future I hope we can have an approach of automatically assigning requests to different org owners, to reduce the amount of time and attention being used, but for now this seems fine :)

<!-- Also keep this in sync with the members of @NixOS/org! -->
- [@infinisil](https://github.com/infinisil)
- [@lassulus](https://github.com/lassulus)
- [@tomberek](https://github.com/tomberek)
- [@winterqt](https://github.com/winterqt)
- [@zimbatm](https://github.com/zimbatm)
## How to contact the team
For any GitHub-related needs, you can reach out to the org owners by either:
- Pinging [@NixOS/org](https://github.com/orgs/NixOS/teams/org)
- [Creating an issue in this repository](https://github.com/NixOS/org/issues/new).
- Messaging in the [Github org owners help desk Matrix room](https://matrix.to/#/#org_owners:nixos.org).

### Authority and processes
This team's role is to manage and unblock users of the github.com/NixOS GitHub organization. The @NixOS/steering gives them autonomy to handle small day-to-day tasks and expects them to escalate bigger decisions.
All org owners can individually take care of implementing:
- Decisions by bodies that have the authority to make GitHub org changes such as:
  - Arbitrary decisions by the Steering Committee.
  - Moderation decisions by the moderation team.
  - Changes from approved RFCs.
- Low-impact changes, such as:

>   - Adding new org members to allow review requests.

Reviewer 3: How do you imagine this to be used? In Nixpkgs, this is handled by [commiter delegation team](https://github.com/NixOS/org/blob/main/doc/github.md#nixpkgs), and in other repos like Nix, I would expect them to have owners who would ideally self-organize.

Author: This is just for the issue of people missing the automated org invite they get after adding themselves to `lib.maintainers`, see https://github.com/NixOS/nixpkgs/issues/234293. An org owner currently needs to resend these missed invites, since there's no automation for that. These invites give no commit access, they're just to allow requesting reviews from maintainers.

  - Creating new unprivileged Nixpkgs teams for mention.
  - Updating repository meta information.

> With approval by at least 2 org owners, they can take care of implementing

Reviewer 3: Maybe make it clearer if the 2 includes the implementer.

```suggestion
With approval by at least 2 other org owners, they can take care of implementing
```

or

```suggestion
With approval by at least one other org owner, they can take care of implementing
```

Author: The latter, I think this should be even more clear:

```suggestion
Org owners need approval from at least one other org owner to take care of implementing
```
